rllib/tuned_examples/marwil/cartpole-bc.yaml

```diff
@@ -1,7 +1,7 @@
 # To generate training data, first run:
 # $ ./train.py --run=PPO --env=CartPole-v0 \
 #      --stop='{"timesteps_total": 50000}' \
-#      --config='{"output": "dataset", "output_config": {"type": "json", "path": "/tmp/out"}, "batch_mode": "complete_episodes"}'
+#      --config='{"output": "dataset", "output_config": {"format": "json", "path": "/tmp/out", "max_num_samples_per_file": 1}, "batch_mode": "complete_episodes"}'
```

Comment:
just to double check, you want to write 1 sample per file??

Reply:
Well 1 sample = 1 episode here so usually has a few hundred timesteps. When I ran this I ended up with 39 files.

Reply:
The main reason is that currently remaining samples are not written at all - they are only written when at least `max_num_samples_per_file` samples are reached during training. I.e. we don't flush remaining results when training is over and I didn't see a great way to make that happen

Reply:
oh this is pretty bad. please add a Note or TODO in dataset_writer.py
this is another case where we are using a workaround to fix underlying problems. and per eric's message, we shouldn't do this long-term.
let's intercept a signal somewhere and flush the last shard before things go down.

Reply:
Ah yeah that's what I mentioned in my first comment to this PR. Ok, will add a todo in the file. I've tried to flush on `__del__` but that doesn't seem to work with actors, so no quick fix. Probably we need to call some kind of deconstructor/cleanup method manually

Reply:
ah, it all comes back. I didn't understand what you meant at the time :)
cool, interesting topic ...